AmplifyPlugins/API/Sources/AWSAPIPlugin/Configuration/AWSAPIEndpointInterceptors.swift

```diff
@@ -16,8 +16,17 @@ struct AWSAPIEndpointInterceptors {
     let apiAuthProviderFactory: APIAuthProviderFactory
     let authService: AWSAuthServiceBehavior?
 
+    /// The order of interceptor decoration is as follows:
```

Comment:
The new prelude / postlude breakdown is great! 

Just a heads up, I think this new order is not b/w compatible with previous order: 

Previously it was:

1/ custom request headers
2/ library defined interceptors (now broken up into prelude and postlude) 
3/ user defined interceptors

Now it's 

1/ prelude interceptors
2/ custom request headers
3/ user defined interceptors
4/ postlude interceptors

Previously prelude interceptors would run after setting custom headers on request.  Now they'll run after running prelude interceptors. 

If this is not a concern, please feel free to disregard.

Reply:
@sandeepklr Thanks for reviewing.

Yes, the behavior would only change if customer is either:

1. Customizing the headers that prelude interceptors are modifying: 
  - In this case, the prelude interceptors primarily decorate the URLRequest by adding specific `x-amz-*` headers to facilitate the proper functioning of the Amplify API plugin with APIGateway. Customers should not predefine these headers when constructing the REST request.

2. Overriding the headers in customer interceptor that postlude interceptors are modifying:
  - In this case, the postlude interceptor is responsible for computing the signature to ensure the integrity of the URLRequest. If the customer calculates the signature within their interceptor, it will be overwritten by the postlude interceptor.

It's worth noting that, in both cases, we maintain the `User-Agent` header for analytics purposes.